src/onegov/feriennet/models/notification_template.py

```diff
@@ -134,8 +119,33 @@ def bind(
         method.__func__.__doc__ = self.request.translate(description)
         self.bound[token] = method
 
-    def render(self, text: str) -> Markup:
-        text = escape(text)
+    def render(self, text: str) -> str:
+        """
+        Escapes and renders the given text, replacing all tokens with the
+        result of the bound methods and 'linkifying' the result.
+
+        :param text:
+        :return: clean text
+        """
+
+        def as_paragraphs(text: str) -> Iterator[str]:
```

Comment:
NOTE: I moved `as_paragraphs` inside `render` method as I reverted using `Markup` and `as_paragraphs` does not return safe html code. I finally use `bleach.clean` to make it safe.

Reply:
I'd go back to the safe `as_paragraphs` function, there's still plenty of ways to add links this way.